examples/amqp/src/main/java/io/cloudevents/examples/amqp/vertx/AmqpClient.java

```diff
@@ -0,0 +1,160 @@
+package io.cloudevents.examples.amqp.vertx;
+
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.qpid.proton.amqp.messaging.Accepted;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.message.Message;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.amqp.ProtonAmqpMessageFactory;
+import io.cloudevents.core.message.MessageReader;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonClientOptions;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonHelper;
+import io.vertx.proton.ProtonMessageHandler;
+import io.vertx.proton.ProtonQoS;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * A example vertx-based AMQP client that interacts with a remote AMQP server to send and receive CloudEvent messages.
+ */
+public class AmqpClient {
+
+    private static ProtonConnection connection;
+
+    private static final String SERVER_HOST = "127.0.0.1";
+    private static final int SERVER_PORT = 5672;
+    private static final String SEND_MESSAGE = "send";
+    private static final String RECEIVE_MESSAGE = "receive";
+
+    private static PrintWriter writer = new PrintWriter(System.out, true);
+
+    public static void main(String args[]) {
+
+        if (args.length < 1) {
+            writer.println("Usage: AmqpClient [send|receive]");
+            return;
+        }
+
+        final String action = args[0].toLowerCase();
+
+        final Vertx vertx = Vertx.vertx();
+
+        switch (action) {
+
+        case SEND_MESSAGE:
+            connectToServer(vertx, SERVER_HOST, SERVER_PORT)
+                    .compose(conn -> {
+                        connection = conn;
+                        writer.printf("[Client] Connected to %s:%s", SERVER_HOST, SERVER_PORT);
+
+                        return openSenderLink();
+                    }).map(sender -> {
```

Comment:
in vert.x 4 you can use `onSuccess` if you need to run something only when the future succeed, but without returning anything